### PR TITLE
Add .init_files.before_run hook (related to issue #562)

### PR DESCRIPTION
**Pull request by felixn**
See issue #562

### CODE REVIEW

File: vunit/ui.py [hunk 1]
```diff
@@ -136,10 +136,17 @@
...
    A list of user defined DO/TCL-files that is sourced after the design has been loaded.
+   They will be executed during ``vunit_load``, after the top level has been loaded
+   using the ``vsim`` command.
    During script evaluation the ``vunit_tb_path`` variable is defined
    as the path of the folder containing the test bench.
    Must be a list of strings.
 
+``modelsim.init_files.before_run``
+   A list of user defined DO/TCL-files that is sourced before the simulation is run.
+   They will be executed at the start of ``vunit_restart``.
```
Comment (kraigher) [inline vunit/ui.py]: This is a bit misleading. The files will be sources in vunit_run which incidentally is called by vunit_restart.

File: vunit/ui.py [hunk 2]
```diff
@@ -158,10 +165,17 @@
...
    A list of user defined DO/TCL-files that is sourced after the design has been loaded.
+   They will be executed during ``vunit_load``, after the top level has been loaded
+   using the ``vsim`` command.
    During script evaluation the ``vunit_tb_path`` variable is defined
    as the path of the folder containing the test bench.
    Must be a list of strings.
 
+``rivierapro.init_files.before_run``
+   A list of user defined DO/TCL-files that is sourced before the simulation is run.
+   They will be executed at the start of ``vunit_restart``.
```
Comment (kraigher) [inline vunit/ui.py]: This is a bit misleading. The files will be sources in vunit_run which incidentally is called by vunit_restart.

File: vunit/ui.py (hunk 1 above)
Comment (felixn) [reply vunit/ui.py]: You're right, I meant to write vunit_run. It's more explicit now, I hope it's clear enough.

File: vunit/ui.py (hunk 2 above)
Comment (felixn) [reply vunit/ui.py]: Same as above.

**Pull request merged**